### DISCUSSION TITLE
Sioyek v1.5 error

### DISCUSSION THREAD

**Discussion by Bogfoot** [ahrm/sioyek#384] [Q&A] [answered]
Greetings, long time user first time writer here.
Not sure if this was adressed before: After downloading the appimage v1.5 for linux
I get core dump messages instantly when running sioyek. Here is the full message I get.
[message.txt](https://github.com/ahrm/sioyek/files/9420077/message.txt)
v1.4. works fine.

Is there something I can do to fix this?

**Comment by ahrm** [comment]
It seems that it can not open `shared.db` file for some reason. Does `/path/.local/share/Sioyek/shared.db` exist?
And make sure sioyek has write access to this file (maybe run sioyek as admin and see if the issue persists).

**Reply by Bogfoot** [reply]
That file exists, along with `auto.config`, `last_document_path.txt` , `local.db` ,  `test.db` , and `tutorial.pdf`. Also, Sioyek is being run with `Exec=/home/bogfoot/Downloads/Sioyek %F` in the `~/.local/share/applications/Sioyek.desktop` file, if that's useful somehow.

**Reply by ahrm** [reply]
Which distribution are you using? This path: `/path/.local/share/Sioyek/local.db` looks very unusual.

**Reply by Bogfoot** [reply]
It's ubuntu 22.04. `/path` = `/home/bogfoot/, I just changed it in the log by habit.

**Reply by ahrm** [reply]
Is your username actually `bogfoot`? I am asking because the log complains about wide characters, I am wondering if you have any non-ascii characters in your username?

**Reply by Bogfoot** [reply]
Yes, it's just Bogfoot. No special characters used.

**Reply by ahrm** [reply]
Does running sioyek directly from the appimage or command line also cause a crash?

**Reply by ahrm** [reply]
Okay I have a theory. Does the file you are trying to open have non-ascii characters (or is in a path with such characters)?

**Reply by Bogfoot** [reply]
> Does running sioyek directly from the appimage or command line also cause a crash?

Yes, running from the command line or file explorer causes an instant crash as well.

> Okay I have a theory. Does the file you are trying to open have non-ascii characters (or is in a path with such characters)?

Neither the file or the entire filepath have non-ascii characters. The full file path is `/home/bogfoot/Desktop/Prezentacije\ za\ Faks/Diplomski/Clanci/Digitalno/Dark\ Matter\ Annual\ Modulation.pdf`.
Sorry for the late reply!

**Comment by ahrm** [comment, marked as answer]
I think the issue should be fixed in d49becfad28854cde7dc078ee3929d625e4009c2. There was a bug where `std::filesystem::exists` would crash when called with non-ascii characters (I am pretty sure it is a bug with gun's cpp implementation). Now we have replaced it with `QFile::exists`.

**Reply by Bogfoot** [reply]
I'll check it out right now and share the results!

**Reply by ahrm** [reply]
If the path of the file doesn't have non-ascii characters, then it is unlikely that this fixed the problem.

One other thing: have you set the value of `shared_database_path` in your `prefs_user.config`? It might be useful if you can share your `prefs_user.config` here.

**Reply by Bogfoot** [reply]
Yes I've set it to `~/.config/sioyek`.
[Bogfoot_prefs_user.txt](https://github.com/ahrm/sioyek/files/9438777/Bogfoot_prefs_user.txt)
Here's my `prefs_user.config` file.

**Reply by ahrm** [reply]
Well, this is wrong. The path should point to a file (for example `/home/bogfoot/.config/sioyek/shared.db`) not a directory.

**Reply by Bogfoot** [reply]
It's a file. Just has no extension. Should it have an extension?

**Reply by ahrm** [reply]
I am not sure, does adding the extension fix the issue?

**Reply by Bogfoot** [reply]
Seems to have fixed it! Guess the extension mattered for 1.5 for some reason.
Thanks for the assist!
